### DISCUSSION TITLE
Setting up code project ai with frigate

### DISCUSSION THREAD

**Discussion by Chris87871** [blakeblackshear/frigate#9347] [Ask A Question]
I tried to set in my config in frigate deepstack as a detectors but spits out an error that it not  a valid detector. I was reading there a plugin possibley I need to install. Is there more documentation to set this up step by step. I can get code project ai docker  fine and edit frigate config there's something I am missing.

**Comment by NickM-27** [comment]
you're probably running the wrong version of frigate, this is only available in 0.13

**Reply by Chris87871** [reply]
Thanks reinstalled with the right branch and it works. Question though is one deepstack detector enough?  For day 3 to 4 cameras?

**Reply by NickM-27** [reply]
it probably wouldn't hurt to setup multiple

**Reply by father1e** [reply]
Could you share your working config? I have 0.13 and latest codeproject:ai but don't get it to create any event.  I am using a custom model. I see in cp:ai logs it detects various objects, ie. fedex, ups; and I have put these as objects in my frigate config. It is not generating events.

**Reply by NickM-27** [reply]
have you added the labels to the frigate config `objects -> track` section?

**Reply by father1e** [reply]
I have two sections of the frigate with the objects
#Global Object Settings
objects:
 track:
  - amazon
  - garbage
  - walmart
  - fedex
  - ups
  - usps
  - Found_USPS
  - USPS
  
And under the camera:
    zones:
      mailbox:
        coordinates: 337,104,340,210,0,214,0,104
        objects:
          - amazon
          - garbage
          - walmart
          - fedex
          - ups
          - usps
          - USPS
          - Found_USPS

Using wireshark I can see cp:ai returning responses, both for no object detected and for found objects.
My frigate is on rpi4 in docker, running 0.13-B3. Is there an RC1 image for docker on rpi?

**Reply by NickM-27** [reply]
yes, seems you are running on quite an old build

**Reply by father1e** [reply]
After further research I find the dev version I have running is the 13.0-rc1 version.

**Reply by father1e** [reply]
I appear to have resolved my issue. Now my frigate with codeproject:ai is generating events.
I needed to create a labelmap.txt file. The documentation doesn't explain the format of this so I simply created a file with the object names, one per line.  I am using a model file in cp:ai named delivery.pt. Can't remember where I found this on the internet. Checking the log in cp:ai I found the object names it was detecting, like fedex, ups, usps.

I copied the labelmap.txt file into the docker container on my RPi where frigate runs. Then added this to my config
#Global Object Settings
model:
  labelmap_path: /labelmap.txt

**Reply by suhl79** [reply]
Hey @father1e I'm currently working on the same issue with exact same model from CP.AI, would you be able to share the mapping file labelmap.txt that you have created for that purpose?

**Comment by Chris87871** [comment]
So far mostly good with deepstack however one camera stops detection after some time and I get this error in log.
error while decoding mb 79 35 byte stream. Any ideas or suggestions.  I restart frigate goes away.
